shared/src/main/scala/magnolify/shared/AnnotationType.scala

```diff
@@ -46,6 +46,6 @@ object AnnotationType {
     val j = q"classOf[${annotated.typeSymbol.asClass}].getAnnotations.toList"
     val annotations = q"_root_.scala.List(..$trees) ++ $j"
 
-    q"_root_.magnolify.shared.AnnotationType[$wtt]($annotations)"
+    q"new _root_.magnolify.shared.AnnotationType[$wtt]($annotations)"
```

Comment:
I sometimes got a compile error because of ambiguous apply call.
Since this is called from `apply[T]` prefer using `new`